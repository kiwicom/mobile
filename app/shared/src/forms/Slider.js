// @flow

import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import type { OnLayout } from '@kiwicom/react-native-app-shared';
import MultiSlider from '@ptomasroos/react-native-multi-slider';

import Color from '../Color';

type Props = {|
  onChange: (number[]) => void,
  min: number,
  max: number,
  startValue: number,
  endValue?: number,
  step?: number,
  snapped?: boolean,
  style?: Object,
|};

type State = {|
  width: number,
|};

export default class Slider extends React.Component<Props, State> {
  state = {
    width: 0,
  };

  onLayout = ({ nativeEvent }: OnLayout) => {
    this.setState({ width: nativeEvent.layout.width });
  };

  render() {
    const values = [this.props.startValue];
    if (this.props.endValue) {
      values.push(this.props.endValue);
    }

    return (
      <View onLayout={this.onLayout} style={styles.sliderWrapper}>
        <MultiSlider
          values={values}
          min={this.props.min}
          max={this.props.max}
          snapped={this.props.snapped}
          allowOverlap
          selectedStyle={styles.selected}
          sliderLength={this.state.width - 2 * SLIDER_PADDING}
          touchDimensions={{
            height: 30,
            width: 30,
            borderRadius: 0,
            slipDisplacement: 500,
          }}
          unselectedStyle={styles.unselected}
          trackStyle={styles.track}
          markerStyle={styles.marker}
          pressedMarkerStyle={styles.marker}
          containerStyle={[styles.container, this.props.style]}
          onValuesChange={this.props.onChange}
        />
      </View>
    );
  }
}

const SLIDER_PADDING = 20;

const styles = StyleSheet.create({
  selected: { backgroundColor: Color.sun },
  unselected: { backgroundColor: Color.grey.$400 },
  track: { borderRadius: 5, height: 2 },
  marker: {
    height: 30,
    width: 30,
    borderRadius: 15,
    backgroundColor: '#fff',
    borderWidth: 1,
    borderColor: Color.grey.$400,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.3,
    shadowRadius: 2,
    elevation: 1, // Android
    margin: 3, // necessary to see the Android elevation properly
  },
  container: {
    justifyContent: 'center',
  },
  sliderWrapper: {
    paddingHorizontal: SLIDER_PADDING,
  },
});
