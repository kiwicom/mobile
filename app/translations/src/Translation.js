// @flow

import * as React from 'react';
import { NativeModules } from 'react-native';
import { Text } from '@kiwicom/react-native-app-shared';

import DefaultVocabulary, { type TranslationKeys } from './DefaultVocabulary';
import CaseTransform, {
  type SupportedTransformationsType,
} from './transformations/CaseTransform';

type Props =
  | {|
      id: TranslationKeys,
      values?: Object,
      textTransform?: SupportedTransformationsType,
    |}
  | {|
      passThrough: ?string | ?number, // may be undefined because of IDX output
      textTransform?: SupportedTransformationsType,
    |};

/**
 * Regular translation using registered keys:
 *
 * ```js
 * <Translation id="SingleHotel.Rating.Stars" />
 * ```
 *
 * So called "pass through" translation (this props is not being translated
 * but passed through the component).
 *
 * ```js
 * <Translation passThrough="★★★★★" />
 * ```
 *
 * It's possible to apply text transformations in both cases:
 *
 * ```js
 * <Translation passThrough="★★★★★" textTransform="uppercase" />
 * <Translation id="SingleHotel.Rating.Stars" textTransform="uppercase" />
 * ```
 */
export default class Translation extends React.Component<Props> {
  // use this property to highlight translations for screenshoting or debugging
  highlightTranslations = false;

  replaceValues = (text: string, values?: Object): string => {
    const VARIABLE_REGEX = /__([0-9]+_)?([a-zA-Z-_]+)__/g;
    return text.replace(
      VARIABLE_REGEX,
      (match, variableIndex, variableName) => {
        // 'variableIndex' is currently being ignored (we don't need it)
        const value = values && values[variableName];
        return value !== undefined ? value : '?';
      },
    );
  };

  render = () => {
    const containerStyle = this.highlightTranslations
      ? {
          backgroundColor: this.props.id ? 'orange' : 'pink',
        }
      : undefined;

    if (this.props.id === undefined) {
      // ID prop is missing so let's assume it's "pass through" translation
      // (property passThrough may be undefined)
      return (
        <Text style={containerStyle}>
          {/* $FlowExpectedError: we do not allow to use 'string' in the 'Text' components but translations are exceptions. */}
          {CaseTransform(
            String(this.props.passThrough),
            this.props.textTransform,
          )}
        </Text>
      );
    }

    const key = this.props.id;
    let translatedString = NativeModules.RNTranslationManager.translate(key);

    if (translatedString === key) {
      // fallback to our dummy vocabulary because native code didn't provide translation
      translatedString = DefaultVocabulary[key];
    }

    if (translatedString === undefined) {
      // Everything failed - native translation even our fallback. This usually
      // means we are using key that does not exist. In this case the only thing
      // we can actually do is to use original key as a text.
      translatedString = key;
    }

    const values = this.props.values || {};
    return (
      <Text style={containerStyle}>
        {/* $FlowExpectedError: we do not allow to use 'string' in the 'Text' components but translations are exceptions. */}
        {CaseTransform(
          this.replaceValues(translatedString, values),
          this.props.textTransform,
        )}
      </Text>
    );
  };
}
