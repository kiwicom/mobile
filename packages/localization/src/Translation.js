// @flow

import * as React from 'react';
import { Text } from '@kiwicom/mobile-shared';
import { translate } from '@kiwicom/rnmodules';

import vocabulary, { type TranslationKeys } from './DefaultVocabulary';
import CaseTransform, {
  type SupportedTransformationsType,
} from './transformations/CaseTransform';
import { getTranslation, replaceValues } from './TranslationHelpers';

// Preload translations
Object.keys(vocabulary).forEach(async key => {
  const nativeKey = 'mobile.' + key;

  const translatedString = await translate(nativeKey);
  if (translatedString != null) {
    vocabulary[key] = getTranslation(translatedString, key);
  }
});

type CommonProps = {|
  testID?: string,
  textTransform?: SupportedTransformationsType,
|};

type Props =
  | {|
      id: TranslationKeys,
      values?: Object,
      ...CommonProps,
    |}
  | {|
      passThrough: ?string | ?number, // may be undefined because of IDX output
      ...CommonProps,
    |};

/**
 * Regular translation using registered keys:
 *
 * ```js
 * <Translation id="SingleHotel.Rating.Stars" />
 * ```
 *
 * So called "pass through" translation (this props is not being translated
 * but passed through the component).
 *
 * ```js
 * <Translation passThrough="★★★★★" />
 * ```
 *
 * It's possible to apply text transformations in both cases:
 *
 * ```js
 * <Translation passThrough="★★★★★" textTransform="uppercase" />
 * <Translation id="SingleHotel.Rating.Stars" textTransform="uppercase" />
 * ```
 */
export default class Translation extends React.Component<Props> {
  // use this property to highlight translations for screenshoting or debugging
  highlightTranslations = false;

  render = () => {
    const containerStyle = this.highlightTranslations
      ? {
          backgroundColor: this.props.id ? 'orange' : 'pink',
        }
      : undefined;

    if (this.props.id === undefined) {
      // ID prop is missing so let's assume it's "pass through" translation
      // (property passThrough may be undefined)
      return (
        <Text style={containerStyle}>
          {/* $FlowExpectedError: we do not allow to use 'string' in the 'Text' components but translations are exceptions. */}
          {CaseTransform(
            String(this.props.passThrough),
            this.props.textTransform,
          )}
        </Text>
      );
    }

    const values = this.props.values || {};
    return (
      <Text style={containerStyle}>
        {/* $FlowExpectedError: we do not allow to use 'string' in the 'Text' components but translations are exceptions. */}
        {CaseTransform(
          replaceValues(vocabulary[this.props.id], values),
          this.props.textTransform,
        )}
      </Text>
    );
  };
}
