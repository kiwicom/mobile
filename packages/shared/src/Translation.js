// @flow

import * as React from 'react';
import {
  CaseTransform,
  getTranslation,
  type SupportedTransformationsType,
  type TranslationKeysType,
} from '@kiwicom/mobile-localization';

import Text from './Text';

type CommonProps = {|
  testID?: string,
  textTransform?: SupportedTransformationsType,
|};

type Props =
  | {|
      id: TranslationKeysType,
      values?: Object,
      ...CommonProps,
    |}
  | {|
      passThrough: ?string | ?number, // may be undefined because of optional chaining
      ...CommonProps,
    |};

/**
 * Regular translation using registered keys:
 *
 * ```js
 * <Translation id="SingleHotel.Rating.Stars" />
 * ```
 *
 * So called "pass through" translation (this props is not being translated
 * but passed through the component).
 *
 * ```js
 * <Translation passThrough="★★★★★" />
 * ```
 *
 * It's possible to apply text transformations in both cases:
 *
 * ```js
 * <Translation passThrough="★★★★★" textTransform="uppercase" />
 * <Translation id="SingleHotel.Rating.Stars" textTransform="uppercase" />
 * ```
 */

// use this property to highlight translations for screenshoting or debugging
const highlightTranslations = false;
export default function Translation(props: Props) {
  const containerStyle = highlightTranslations
    ? {
        backgroundColor: props.id ? 'orange' : 'pink',
      }
    : undefined;
  if (props.id === undefined && props.passThrough != null) {
    return (
      <Text style={containerStyle}>
        {/* $FlowExpectedError: we do not allow to use 'string' in the 'Text' components but translations are exceptions. */}
        {CaseTransform(String(props.passThrough), props.textTransform)}
      </Text>
    );
  } else if (props.id === undefined) {
    return null;
  }

  const values = props.values || {};
  return (
    <Text style={containerStyle}>
      {/* $FlowExpectedError: we do not allow to use 'string' in the 'Text' components but translations are exceptions. */}
      {CaseTransform(getTranslation(props.id, values), props.textTransform)}
    </Text>
  );
}
