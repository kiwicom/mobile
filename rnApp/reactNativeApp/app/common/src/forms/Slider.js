// @flow

import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import type { OnLayout } from '@kiwicom/react-native-app-common';
import MultiSlider from '@ptomasroos/react-native-multi-slider';

import { POPUP_PADDING } from '../popup/Popup';
import Color from '../Color';

type Props = {|
  onChange: (number[]) => void,
  min: number,
  max: number,
  startValue: number,
  endValue?: number,
  step?: number,
  // snapped?: boolean, FIXME https://github.com/ptomasroos/react-native-multi-slider/issues/49
  style?: Object,
|};

type State = {|
  width: number,
|};

export default class Slider extends React.Component<Props, State> {
  state = {
    width: 0,
  };

  onLayout = ({ nativeEvent }: OnLayout) => {
    this.setState({ width: nativeEvent.layout.width });
  };

  render() {
    const values = [this.props.startValue];
    if (this.props.endValue) {
      values.push(this.props.endValue);
    }

    return (
      <View onLayout={this.onLayout} style={styles.sliderWrapper}>
        <MultiSlider
          values={values}
          min={this.props.min}
          max={this.props.max}
          allowOverlap
          selectedStyle={styles.selected}
          sliderLength={this.state.width - 2 * POPUP_PADDING}
          touchDimensions={{
            height: 30,
            width: 30,
            borderRadius: 0,
            slipDisplacement: 500,
          }}
          unselectedStyle={styles.unselected}
          trackStyle={styles.track}
          markerStyle={styles.marker}
          containerStyle={[styles.container, this.props.style]}
          onValuesChange={this.props.onChange}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  selected: { backgroundColor: Color.sun },
  unselected: { backgroundColor: Color.grey.$400 },
  track: { borderRadius: 5, height: 2 },
  marker: {
    height: 30,
    width: 30,
    borderRadius: 15,
    backgroundColor: '#fff',
    borderWidth: 0.5,
    borderColor: Color.grey.$400,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.3,
    shadowRadius: 2,
  },
  container: {
    justifyContent: 'center',
  },
  sliderWrapper: {
    paddingHorizontal: POPUP_PADDING,
  },
});
