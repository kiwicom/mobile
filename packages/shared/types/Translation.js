// @flow

import * as React from 'react';

import Translation from '../src/Translation';
import TranslationFragment from '../src/TranslationFragment';

export type TranslationType =
  | React.Element<typeof Translation>
  | React.Element<typeof TranslationFragment>;
