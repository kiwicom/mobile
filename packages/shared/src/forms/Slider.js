// @flow

import * as React from 'react';
import { View } from 'react-native';
import MultiSlider from '@ptomasroos/react-native-multi-slider';
import { defaultTokens } from '@kiwicom/mobile-orbit';

import SliderMarker from './SliderMarker';
import { type OnLayout } from '../../types/Events';
import StyleSheet from '../PlatformStyleSheet';
import { type TranslationType } from '../../types/Translation';
import Price from '../Price';
import Text from '../Text';

type Props = {|
  +onChange: (number[]) => void,
  +min: number,
  +max: number,
  +startValue: number,
  +endValue?: number,
  +step?: number,
  +snapped?: boolean,
  +style?: Object,
  +startLabel?: TranslationType | React.Element<typeof Price>,
  +endLabel?: TranslationType | React.Element<typeof Price>,
|};

type State = {|
  width: number,
|};

export default class Slider extends React.Component<Props, State> {
  state = {
    width: 0,
  };

  onLayout = ({ nativeEvent }: OnLayout) => {
    this.setState({ width: nativeEvent.layout.width });
  };

  getMaxMinAndEnabled = () => {
    const { max, min } = this.props;

    if (max === min && max === this.props.startValue && max === this.props.endValue) {
      // If all values are equal, expand max and min values by one
      // to make dot appear on the center and disable sliding.
      return {
        max: max + 1,
        min: min - 1,
        enabledOne: false,
        enabledTwo: false,
      };
    }

    return {
      max,
      min,
      enabledOne: true,
      enabledTwo: true,
    };
  };

  render() {
    const { max, min, enabledOne, enabledTwo } = this.getMaxMinAndEnabled();
    const values = [this.props.startValue];

    if (this.props.endValue) {
      values.push(this.props.endValue);
    }

    const showLabels = this.props.startLabel != null || this.props.endLabel != null;

    return (
      <View onLayout={this.onLayout} style={styles.sliderWrapper}>
        <MultiSlider
          values={values}
          min={min}
          max={max}
          snapped={this.props.snapped}
          allowOverlap={true}
          selectedStyle={styles.selected}
          sliderLength={this.state.width}
          touchDimensions={{
            height: 30,
            width: 30,
            borderRadius: 0,
            slipDisplacement: 500,
          }}
          unselectedStyle={styles.unselected}
          trackStyle={styles.track}
          containerStyle={[styles.container, this.props.style]}
          onValuesChange={this.props.onChange}
          enabledOne={enabledOne}
          enabledTwo={enabledTwo}
          customMarker={SliderMarker}
        />
        {showLabels && (
          <View style={styles.labelsContainer} testID="labelsContainer">
            {this.props.startLabel != null && (
              <Text style={styles.label}>{this.props.startLabel}</Text>
            )}
            {this.props.endLabel != null && <Text style={styles.label}>{this.props.endLabel}</Text>}
          </View>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  selected: {
    backgroundColor: defaultTokens.paletteBlueNormal,
  },
  unselected: {
    backgroundColor: defaultTokens.paletteInkLighter,
  },
  track: {
    borderRadius: 5,
    height: 4,
  },
  container: {
    justifyContent: 'center',
  },
  labelsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginHorizontal: -10,
  },
  label: {
    color: defaultTokens.colorTextSecondary,
    fontSize: parseInt(defaultTokens.fontSizeTextSmall, 10),
  },
});
