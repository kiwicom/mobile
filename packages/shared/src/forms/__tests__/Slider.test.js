// @flow

import * as React from 'react';
import { PlaygroundRenderer } from '@kiwicom/mobile-playground';
import { create } from 'react-test-renderer';

import Translation from '../../Translation';
import Slider from '../Slider';

const onChange = () => {};

describe('Slider', () => {
  it('renders', () => {
    PlaygroundRenderer.render(
      <Slider
        startValue={1}
        endValue={50}
        onChange={onChange}
        min={1}
        max={50}
      />,
    );
  });

  it('renders correctly with all values equal', () => {
    PlaygroundRenderer.render(
      <Slider
        startValue={500}
        endValue={500}
        onChange={onChange}
        min={500}
        max={500}
      />,
    );
  });

  it('renders with endValue undefined', () => {
    PlaygroundRenderer.render(
      <Slider startValue={1} onChange={onChange} min={1} max={50} />,
    );
  });
});

it('renders one label correctly', () => {
  const wrapper = create(
    <Slider
      startValue={500}
      endValue={500}
      onChange={onChange}
      min={500}
      max={500}
      startLabel={<Translation passThrough="startLabel" />}
    />,
  );

  expect(
    wrapper.root.findByProps({ testID: 'labelsContainer' }),
  ).not.toBeNull();
  expect(wrapper.root.findByType(Translation).props.passThrough).toBe(
    'startLabel',
  );
});

it('renders two labels correctly', () => {
  const wrapper = create(
    <Slider
      startValue={500}
      endValue={500}
      onChange={onChange}
      min={500}
      max={500}
      startLabel={<Translation testID="startLabel" passThrough="startLabel" />}
      endLabel={<Translation testID="endLabel" passThrough="endLabel" />}
    />,
  );

  expect(
    wrapper.root.findByProps({ testID: 'labelsContainer' }),
  ).not.toBeNull();

  expect(wrapper.root.findByProps({ testID: 'startLabel' })).not.toBeNull();
  expect(wrapper.root.findByProps({ testID: 'endLabel' })).not.toBeNull();
});

it('renders correctly without labels', () => {
  const wrapper = create(
    <Slider
      startValue={500}
      endValue={500}
      onChange={onChange}
      min={500}
      max={500}
    />,
  );

  expect(() => {
    wrapper.root.findByProps({ testID: 'labelsContainer' });
  }).toThrow();
});
