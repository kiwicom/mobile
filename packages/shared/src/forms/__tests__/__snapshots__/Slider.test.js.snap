// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders 1`] = `
<Component
  onLayout={[Function]}
  style={
    Object {
      "paddingHorizontal": 20,
    }
  }
>
  <MultiSlider
    allowOverlap={true}
    containerStyle={
      Array [
        Object {
          "justifyContent": "center",
        },
        undefined,
      ]
    }
    customMarker={[Function]}
    customMarkerLeft={[Function]}
    customMarkerRight={[Function]}
    enabledOne={true}
    enabledTwo={true}
    markerOffsetX={0}
    markerOffsetY={0}
    markerStyle={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#bdbdbd",
        "borderRadius": 15,
        "borderWidth": 1,
        "elevation": 1,
        "height": 30,
        "margin": 3,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 2,
        "width": 30,
      }
    }
    max={50}
    min={1}
    onToggleOne={undefined}
    onToggleTwo={undefined}
    onValuesChange={[Function]}
    onValuesChangeFinish={[Function]}
    onValuesChangeStart={[Function]}
    pressedMarkerStyle={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#bdbdbd",
        "borderRadius": 15,
        "borderWidth": 1,
        "elevation": 1,
        "height": 30,
        "margin": 3,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 2,
        "width": 30,
      }
    }
    selectedStyle={
      Object {
        "backgroundColor": "#f9971e",
      }
    }
    sliderLength={-40}
    snapped={false}
    step={1}
    touchDimensions={
      Object {
        "borderRadius": 0,
        "height": 30,
        "slipDisplacement": 500,
        "width": 30,
      }
    }
    trackStyle={
      Object {
        "borderRadius": 5,
        "height": 2,
      }
    }
    unselectedStyle={
      Object {
        "backgroundColor": "#bdbdbd",
      }
    }
    values={
      Array [
        1,
        50,
      ]
    }
  />
</Component>
`;

exports[`Slider renders correctly with all values equal 1`] = `
<Component
  onLayout={[Function]}
  style={
    Object {
      "paddingHorizontal": 20,
    }
  }
>
  <MultiSlider
    allowOverlap={true}
    containerStyle={
      Array [
        Object {
          "justifyContent": "center",
        },
        undefined,
      ]
    }
    customMarker={[Function]}
    customMarkerLeft={[Function]}
    customMarkerRight={[Function]}
    enabledOne={false}
    enabledTwo={false}
    markerOffsetX={0}
    markerOffsetY={0}
    markerStyle={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#bdbdbd",
        "borderRadius": 15,
        "borderWidth": 1,
        "elevation": 1,
        "height": 30,
        "margin": 3,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 2,
        "width": 30,
      }
    }
    max={501}
    min={499}
    onToggleOne={undefined}
    onToggleTwo={undefined}
    onValuesChange={[Function]}
    onValuesChangeFinish={[Function]}
    onValuesChangeStart={[Function]}
    pressedMarkerStyle={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#bdbdbd",
        "borderRadius": 15,
        "borderWidth": 1,
        "elevation": 1,
        "height": 30,
        "margin": 3,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 2,
        "width": 30,
      }
    }
    selectedStyle={
      Object {
        "backgroundColor": "#f9971e",
      }
    }
    sliderLength={-40}
    snapped={false}
    step={1}
    touchDimensions={
      Object {
        "borderRadius": 0,
        "height": 30,
        "slipDisplacement": 500,
        "width": 30,
      }
    }
    trackStyle={
      Object {
        "borderRadius": 5,
        "height": 2,
      }
    }
    unselectedStyle={
      Object {
        "backgroundColor": "#bdbdbd",
      }
    }
    values={
      Array [
        500,
        500,
      ]
    }
  />
</Component>
`;

exports[`Slider renders with endValue undefined 1`] = `
<Component
  onLayout={[Function]}
  style={
    Object {
      "paddingHorizontal": 20,
    }
  }
>
  <MultiSlider
    allowOverlap={true}
    containerStyle={
      Array [
        Object {
          "justifyContent": "center",
        },
        undefined,
      ]
    }
    customMarker={[Function]}
    customMarkerLeft={[Function]}
    customMarkerRight={[Function]}
    enabledOne={true}
    enabledTwo={true}
    markerOffsetX={0}
    markerOffsetY={0}
    markerStyle={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#bdbdbd",
        "borderRadius": 15,
        "borderWidth": 1,
        "elevation": 1,
        "height": 30,
        "margin": 3,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 2,
        "width": 30,
      }
    }
    max={50}
    min={1}
    onToggleOne={undefined}
    onToggleTwo={undefined}
    onValuesChange={[Function]}
    onValuesChangeFinish={[Function]}
    onValuesChangeStart={[Function]}
    pressedMarkerStyle={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#bdbdbd",
        "borderRadius": 15,
        "borderWidth": 1,
        "elevation": 1,
        "height": 30,
        "margin": 3,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 2,
        "width": 30,
      }
    }
    selectedStyle={
      Object {
        "backgroundColor": "#f9971e",
      }
    }
    sliderLength={-40}
    snapped={false}
    step={1}
    touchDimensions={
      Object {
        "borderRadius": 0,
        "height": 30,
        "slipDisplacement": 500,
        "width": 30,
      }
    }
    trackStyle={
      Object {
        "borderRadius": 5,
        "height": 2,
      }
    }
    unselectedStyle={
      Object {
        "backgroundColor": "#bdbdbd",
      }
    }
    values={
      Array [
        1,
      ]
    }
  />
</Component>
`;
